src/app/record/components/work-stack-group/modals/work-bibtex-modal/work-bibtex-modal.component.ts

```diff
@@ -48,49 +50,58 @@ export class WorkBibtexModalComponent implements OnInit, OnDestroy {
       const that = this
 
       reader.onloadend = function (e) {
-        const parsed = bibtexParse.toJSON(reader.result)
-        if (
-          typeof parsed === 'string' &&
-          parsed.substring(0, 5).toLowerCase().indexOf('error') > -1
-        ) {
-          that.bibtexParsingErrorText = parsed
-          that.bibtexParsingError = true
-        } else {
-          const newWorks = []
-          if (parsed.length === 0) {
-            that.bibtexParsingErrorText = that.ngOrcidErrorParsingBibtex
-            that.bibtexParsingError = true
+        let parsed = null
+        try {
+          parsed = bibtexParse.toJSON(reader.result)
+          if (
+            typeof parsed === 'string' &&
+            parsed.substring(0, 5).toLowerCase().indexOf('error') > -1
+          ) {
+            that.bibtexErrorParsingText = parsed
+            that.bibtexErrorParsing = true
             that.loadingWorks = false
-          }
-          while (parsed.length > 0) {
-            const cur = parsed.shift()
-            const bibtexEntry = cur.entryType.toLowerCase()
-            if (bibtexEntry !== 'preamble' && bibtexEntry !== 'comment') {
-              // Filtering @PREAMBLE and @COMMENT
-              newWorks.push(that.populateWork(cur))
-            }
-          }
-          if (newWorks.length > 0) {
-            that._recordWorksService
-              .worksValidate(newWorks)
-              .pipe(first())
-              .subscribe((data) => {
-                data.forEach((work) => {
-                  that.worksFromBibtex.push(work)
-                })
-                that.worksFromBibtex.forEach((w) => {
-                  const newPutCode = 'new-' + that.addedWorkCount++
-                  w.putCode = {
-                    value: newPutCode,
-                  }
-                  that.group[newPutCode] = new FormGroup({
-                    checked: new FormControl(false),
-                  })
-                })
-                that.importForm = new FormGroup(that.group)
+          } else {
+            if (parsed) {
+              const newWorks = []
+              if (parsed?.length === 0) {
```

Comment:
```suggestion
              if (parsed.length === 0) {
```

The parse is truthy based on the wrapping `if (parsed)`